src/frontend/frontend/views/story_views.py

```
@@ -451,22 +520,141 @@ def get_search_dialog(cls) -> tuple[str, int]:
+            if make_default:
+                default_id = matching_filter["id"] if matching_filter else saved_filters[-1]["id"]
+                for saved_filter in saved_filters:
+                    saved_filter["is_default"] = saved_filter["id"] == default_id
+
+            response = cls._update_saved_filters_profile(saved_filters)
+            if getattr(response, "ok", False):
+                session[ASSESS_SAVED_FILTER_SESSION_KEY] = make_default
+            return cls._render_saved_filters_mutation_response(response, "Assess filter saved.", saved_filters)
         except HTTPException:
             raise
         except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)